src/main/kotlin/com/team4099/robot2023/subsystems/intake/IntakeIO.kt

```diff
@@ -0,0 +1,111 @@
+
+      table?.put("groundIntakeLeaderArmStatorCurrentAmps", leaderArmStatorCurrent.inAmperes)
+      table?.put("groundIntakeFollowerArmStatorCurrentAmps", followerArmStatorCurrent.inAmperes)
+
+      table?.put("groundIntakeLeaderTempCelcius", leaderTempCelcius.inCelsius)
+      table?.put("groundIntakeFollowerTempCelcius", followerTempCelcius.inCelsius)
+
+      table?.put("groundIntakeRollerApplied", rollerApplied)
+
+      table?.put("groundIntakeRollerSupplyCurrentAmps", rollerSupplyCurrent.inAmperes)
+      table?.put("groundIntakeRollerStatorCurrentAmps", rollerStatorCurrent.inAmperes)
+
+      table?.put("groundIntakeRollerTempCelcius", rollerTempCelcius.inCelsius)
+    }
+
+    override fun fromLog(table: LogTable?) {
+      table?.getDouble("groundIntakePositionDegrees", leaderArmPosition.inDegrees)?.let {
+        leaderArmPosition = it.degrees
+      }
+      table?.getDouble("groundIntakeVelocityDegreesPerSec", leaderArmVelocity.inDegreesPerSecond)
+        ?.let { leaderArmVelocity = it.degrees.perSecond }
+      table?.getDouble("groundIntakeLeaderArmAppliedOutput", leaderArmAppliedOutput)?.let {
+        leaderArmAppliedOutput = it
+      }
+      table?.getDouble("groundIntakeFollowerArmAppliedOutput", followerArmAppliedOutput)?.let {
+        followerArmAppliedOutput = it
+      }
+      table?.getDouble("groundIntakeLeaderArmSupplyCurrentAmps", leaderArmSupplyCurrent.inAmperes)
+        ?.let { leaderArmSupplyCurrent = it.amps }
+      table?.getDouble(
+        "groundIntakeFollowerArmSupplyCurrentAmps", followerArmSupplyCurrent.inAmperes
+      )
+        ?.let { followerArmSupplyCurrent = it.amps }
+      table?.getDouble("groundIntakeLeaderArmStatorCurrentAmps", leaderArmStatorCurrent.inAmperes)
+        ?.let { leaderArmStatorCurrent = it.amps }
+      table?.getDouble(
+        "groundIntakeFollowerArmStatorCurrentAmps", followerArmStatorCurrent.inAmperes
+      )
+        ?.let { followerArmStatorCurrent = it.amps }
+      table?.getDouble("groundIntakeLeaderTempCelcius", leaderTempCelcius.inCelsius)?.let {
+        leaderTempCelcius = it.celsius
+      }
+      table?.getDouble("groundIntakeFollowerTempCelcius", followerTempCelcius.inCelsius)?.let {
+        followerTempCelcius = it.celsius
+      }
+      table?.getDouble("groundIntakeRollerApplied", rollerApplied)?.let { rollerApplied = it }
+      table?.getDouble("groundIntakeRollerSupplyCurrentAmps", rollerSupplyCurrent.inAmperes)?.let {
+        rollerSupplyCurrent = it.amps
+      }
+      table?.getDouble("groundIntakeRollerStatorCurrentAmps", rollerStatorCurrent.inAmperes)?.let {
+        rollerStatorCurrent = it.amps
+      }
+      table?.getDouble("groundIntakeRollerTempCelcius", rollerTempCelcius.inCelsius)?.let {
+        rollerStatorCurrent = it.amps
+      }
+    }
+  }
+
+  fun updateInputs(inputs: IntakeIOInputs) {}
+
+  fun setRollerPower(outputPower: Double) {}
```

Comment:
Do we want to do this or do we just want to take a voltage here? Ideally we can get rid of the percent outputs pretty much everywhere to make the behavior more predictable (so that the behavior is at least potentially the same when we are browning out vs. when the battery is new)